### ISSUE TITLE
Object#equals implementations

### ISSUE THREAD

**Issue by Sxtanna** [kapoorlabs/kiara#14]
I noticed in a couple of your domain classes, you implemented `Object#equals` with the following pattern:
```java
Type other = (Type) obj;

if (value == null) {
	if (other.value != null)
		return false;
} else if (!value.equals(other.value))
	return false;
// ... other checks

return true;
```

Is there an actual reason you're doing this? This pattern of checks is perfectly satisfied by the 
`java.util.Objects.equals(Object a, Object b)` static method provided by the JDK, and would be infinitely more readable and maintainable.

ex. `Range.java`
```java
@Override
public boolean equals(Object obj) {
    if (this == obj)
        return true;
    if (!(obj instanceof Range))
        return false;
    Range other = (Range) obj;
    if (lowerLimit == null) {
        if (other.lowerLimit != null)
            return false;
    } else if (!lowerLimit.equals(other.lowerLimit))
        return false;
    if (prefix == null) {
        if (other.prefix != null)
            return false;
    } else if (!prefix.equals(other.prefix))
        return false;
    if (upperLimit == null) {
        if (other.upperLimit != null)
            return false;
    } else if (!upperLimit.equals(other.upperLimit))
        return false;
    return true;
}
```
would be
```java
@Override
public boolean equals(Object obj) {
    if (this == obj)
        return true;
    if (!(obj instanceof Range))
        return false;
    Range other = (Range) obj;
    return Objects.equals(lowerLimit, other.lowerLimit) &&
           Objects.equals(prefix, other.prefix) &&
           Objects.equals(upperLimit, other.upperLimit);
}
```

much less code to read through and understand each condition.

**Comment by Sxtanna** [comment]
ref: `Objects.equals`
```java
public static boolean equals(Object a, Object b) {
    return (a == b) || (a != null && a.equals(b));
}
```

**Comment by kapoor-labs** [comment]
This sounds good! There wasn't any specific reason of writing the verbose code. Since JDK provided library function does all the checks, we can use that and reduce verbosity!

Thanks for reporting this!

**Comment by Sxtanna** [comment]
Addressed in #15

**Comment by kapoor-labs** [comment]
Fix in development.
Will close the issue once patch is released.
Since this fixes readability of code, it is not urgent to release a patch.

**Comment by kapoor-labs** [comment]
Released version 2.0.12

The released version also contains support for enum data types.
